src/main/java/com/zendesk/maxwell/util/AbstractConfig.java

```diff
@@ -95,6 +100,29 @@ protected Properties readPropertiesFile(String filename, Boolean abortOnMissing)
 		return p;
 	}
 
+	protected Properties readPropertiesEnv(String envConfig) {
+		LOGGER.debug("Attempting to read env_config param: {}", envConfig);
+		String envConfigJsonRaw = System.getenv(envConfig);
+		if (envConfigJsonRaw != null && envConfigJsonRaw.startsWith("{")) {
```

Comment:
might want to be forgiving of whitespace?

Reply:
added `trim()` to cover whitespace including newline, carriage return, spaces, tabs